Index: code-server/lib/vscode/src/vs/platform/actions/browser/dropdownWithPrimaryActionViewItem.ts
===================================================================
--- code-server.orig/lib/vscode/src/vs/platform/actions/browser/dropdownWithPrimaryActionViewItem.ts
+++ code-server/lib/vscode/src/vs/platform/actions/browser/dropdownWithPrimaryActionViewItem.ts
@@ -94,8 +94,9 @@ export class DropdownWithPrimaryActionViewItem extends BaseActionViewItem {
 
 	override render(container: HTMLElement): void {
 		this._container = container;
-		super.render(this._container);
+		this.element = this._container;
 		this._container.classList.add('monaco-dropdown-with-primary');
+		this.updateEnabled();
 		const primaryContainer = DOM.$('.action-container');
 		primaryContainer.role = 'button';
 		primaryContainer.ariaDisabled = String(!this.action.enabled);
